### ISSUE TITLE
GAE and returns run over padding because neither function takes an action mask

### ISSUE THREAD

**Issue by CodersAcademy006** [ellanorai/ThinkRL#99]
`thinkrl/utils/metrics.py:173` `compute_advantages` and `:199` `compute_returns` take no mask and no `dones` argument:

```python
def compute_advantages(rewards, values, gamma=0.99, lambda_=0.95, normalize=True):
    batch_size, seq_len = rewards.shape
    ...
    gae = 0
    for t in reversed(range(seq_len)):
        gae = deltas[:, t] + gamma * lambda_ * gae
        advantages[:, t] = gae
```

The recursion starts at `t = seq_len - 1`, which for any sequence shorter than the batch maximum is padding. Whatever the value head emits on those slots enters `deltas` and propagates backwards into the real tokens.

Reproduced with one sequence of true length 2 in a padded width of 3, rewards `[0, 5, 0]`:

```
pad-slot value 7.0  ->  advantages [5.028, 5.347, -7.000]
pad-slot value 0.0  ->  advantages [4.703, 5.000,  0.000]
```

The first two positions are real tokens and their advantages differ by roughly 7% purely as a function of what the critic happened to predict on a pad position. The value head is not trained on pad slots, so that prediction is arbitrary and will drift over training.

`PPOAlgorithm.train_on_rollout` builds `dense_rewards = torch.zeros_like(old_values)` at `ppo.py:216` over the full padded width and passes it straight in, so PPO is affected on every non-uniform batch. `thinkrl/algorithms/vapo.py:156` `compute_adaptive_gae` is a separate copy of the same recursion with the same gap.

**Suggested fix**

Add an `action_mask` argument to both functions and zero `deltas` at masked positions before the backward pass, so padding contributes nothing to the accumulator. `attention_mask` and the `labels != -100` mask are both already available at the call sites.

I am happy to open a PR for this.
